inference/core/workflows/core_steps/models/foundation/segment_anything/v1.py

```diff
@@ -0,0 +1,293 @@
+        kind=[
+            BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
+            BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND,
+            BATCH_OF_KEYPOINT_DETECTION_PREDICTION_KIND,
+        ]
+    ) = Field(  # type: ignore
+        description="Boxes (from other model predictions)",
+        examples=["$steps.object_detection_model.predictions"],
+        default=None,
+    )
+
+    sam2_model: Union[
+        WorkflowParameterSelector(kind=[STRING_KIND]),
+        Literal["hiera_large", "hiera_small", "hiera_tiny", "hiera_b_plus"],
+    ] = Field(
+        default="hiera_large",
+        description="Model to be used.  One of hiera_large, hiera_small, hiera_tiny, hiera_b_plus",
+        examples=["hiera_large", "$inputs.openai_model"],
+    )
+
+    @classmethod
+    def accepts_batch_input(cls) -> bool:
+        return True
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(
+                name="predictions",
+                kind=[BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND],
+            ),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class SegmentAnything2BlockV1(WorkflowBlock):
+
+    def __init__(
+        self,
+        model_manager: ModelManager,
+        api_key: Optional[str],
+        step_execution_mode: StepExecutionMode,
+    ):
+        self._model_manager = model_manager
+        self._api_key = api_key
+        self._step_execution_mode = step_execution_mode
+
+    @classmethod
+    def get_init_parameters(cls) -> List[str]:
+        return ["model_manager", "api_key", "step_execution_mode"]
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    def run(
+        self,
+        images: Batch[WorkflowImageData],
+        sam2_model: str,
+        boxes: Batch[sv.Detections],
+    ) -> BlockResult:
+        if self._step_execution_mode is StepExecutionMode.LOCAL:
+            return self.run_locally(images=images, sam2_model=sam2_model, boxes=boxes)
+        elif self._step_execution_mode is StepExecutionMode.REMOTE:
+            raise NotImplementedError(
+                "Remote execution is not supported for Segment Anything."
+            )
+        else:
+            raise ValueError(
+                f"Unknown step execution mode: {self._step_execution_mode}"
+            )
+
+    def run_locally(
+        self,
+        images: Batch[WorkflowImageData],
+        sam2_model: str,
+        boxes: Batch[sv.Detections],
+    ) -> BlockResult:
+
+        predictions = []
+
+        if not boxes:
+            boxes = [None] * len(images)
+
+        for single_image, boxes_for_image in zip(images, boxes):
+            prompt_class_ids = []
+            prompt_class_names = []
+            prompt_index = 0
+
+            prompts = Sam2PromptSet()
+            if boxes_for_image is not None:
```

Comment:
block prevents empty inputs so effectively this if is not needed

Reply:
need the check here, because we potentially padded this on line 162 though or you are saying the EE will always provided array with None already and we can scrap 162?